pkg/registry/flowcontrol/rest/storage_flowcontrol.go

```diff
@@ -0,0 +1,148 @@
+	storage["prioritylevelconfigurations"] = priorityLevelConfigurationStorage
+	storage["prioritylevelconfigurations/status"] = priorityLevelConfigurationStatusStorage
+
+	return storage
+}
+
+func (p RESTStorageProvider) PostStartHook() (string, genericapiserver.PostStartHookFunc, error) {
+	systemPreset := SystemPresetData{
+		FlowSchemas:                 bootstrap.SystemFlowSchemas(),
+		PriorityLevelConfigurations: bootstrap.SystemPriorityLevelConfigurations(),
+	}
+	// TODO: default flow-schemas and priority levels
+	return PostStartHookName, systemPreset.EnsureSystemPresetConfiguration(), nil
+}
+
+type SystemPresetData struct {
+	FlowSchemas                 []*flowcontrolv1alpha1.FlowSchema
+	PriorityLevelConfigurations []*flowcontrolv1alpha1.PriorityLevelConfiguration
+}
+
+func (d SystemPresetData) EnsureSystemPresetConfiguration() genericapiserver.PostStartHookFunc {
+	return func(hookContext genericapiserver.PostStartHookContext) error {
+		flowcontrolClientSet := flowcontrolclient.NewForConfigOrDie(hookContext.LoopbackClientConfig)
+		// Adding system priority classes is important. If they fail to add, many critical system
+		// components may fail and cluster may break.
+		err := wait.Poll(1*time.Second, 30*time.Second, func() (done bool, err error) {
+			if err != nil {
+				utilruntime.HandleError(fmt.Errorf("unable to initialize client: %v", err))
+				return false, nil
+			}
+
+			for _, flowSchema := range d.FlowSchemas {
+				_, err := flowcontrolClientSet.FlowSchemas().Get(flowSchema.Name, metav1.GetOptions{})
+				if err != nil {
+					if apierrors.IsNotFound(err) {
+						_, err := flowcontrolClientSet.FlowSchemas().Create(flowSchema)
+						if err != nil && !apierrors.IsAlreadyExists(err) {
+							return false, err
+						} else if err == nil {
+							klog.V(3).Infof("created system preset FlowSchema %s", flowSchema.Name)
+						} else {
+							klog.V(3).Infof("system preset FlowSchema %s already exists, skipping creating", flowSchema.Name)
+						}
+					} else {
+						// Unable to get the priority class for reasons other than "not found".
+						klog.Warningf("unable to get FlowSchema %v: %v. Retrying...", flowSchema.Name, err)
+						return false, nil
+					}
+				}
+			}
+			for _, priorityLevelConfiguration := range d.PriorityLevelConfigurations {
+				_, err := flowcontrolClientSet.PriorityLevelConfigurations().Get(priorityLevelConfiguration.Name, metav1.GetOptions{})
+				if err != nil {
+					if apierrors.IsNotFound(err) {
+						_, err := flowcontrolClientSet.PriorityLevelConfigurations().Create(priorityLevelConfiguration)
+						if err != nil && !apierrors.IsAlreadyExists(err) {
+							return false, err
+						} else if err == nil {
+							klog.V(3).Infof("created system preset PriorityLevelConfiguration %s", priorityLevelConfiguration.Name)
+						} else {
+							klog.V(3).Infof("system preset PriorityLevelConfiguration %s already exists, skipping creating", priorityLevelConfiguration.Name)
```

Comment:
I think it will be confusing for admins to see a message saying "already exists" in rare circumstances.  Remember, the more common "already exists" scenario above provokes no logging.  The difference between that scenario and this one is concurrent activity.  The message that is logged here does not really explain what is distinctive about this situation.